File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0, right = numbers.size() - 1;
        while (left < right) {
            int sum = numbers[left] + numbers[right];
            if (sum == target) {
                return {left + 1, right + 1}; // 1-indexed
            } else if (sum < target) {
                left++;  // need a larger sum
            } else {
                right--; // need a smaller sum
            }
        }
        return {-1, -1};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let left = 0;
    let right = numbers.length - 1;
    let res = [];
    while(left < right){
        const sum = numbers[left] + numbers[right];

        if(sum === target){
            return [left + 1, right + 1]
        }else if(sum > target){
            right--;
        }else{
            left++;
        }
    }

    return [-1, -1];
};